src/controllers/api-key.js

```diff
@@ -10,52 +10,181 @@
+ * @param context.env.API_KEY_CONFIGURATION - Configuration for the API key.
  * @returns {Object} ApiKey Controller
  * @constructor
  */
 function ApiKeyController(context) {
-  // eslint-disable-next-line no-unused-vars
-  const { log } = context;
+  const {
+    dataAccess, log, env, attributes, imsClient,
+  } = context;
+
+  let apiKeyConfiguration = {};
+  try {
+    apiKeyConfiguration = JSON.parse(env.API_KEY_CONFIGURATION);
+  } catch (error) {
+    log.error(`Failed to parse API Key configuration: ${error.message}`);
+  }
+
+  const { maxDomainsPerApiKey = 1, maxApiKeys = 3 } = apiKeyConfiguration;
   const HEADER_ERROR = 'x-error';
 
   function createErrorResponse(error) {
-    return createResponse({}, error.status, {
+    return createResponse({}, error.status || SERVER_ERROR, {
       [HEADER_ERROR]: error.message,
     });
   }
 
-  // eslint-disable-next-line no-unused-vars
+  /**
+   * Validate the request data.
+   * @param data
+   */
   function validateRequestData(data) {
-    // Not implemented yet
+    if (!isObject(data)) {
+      throw new ErrorWithStatusCode('Invalid request: missing application/json request data', STATUS_BAD_REQUEST);
+    }
+
+    if (!Array.isArray(data.features) || data.features.length === 0) {
+      throw new ErrorWithStatusCode('Invalid request: missing features in the request data', STATUS_BAD_REQUEST);
+    }
+
+    if (!Array.isArray(data.domains) || data.domains.length === 0) {
+      throw new ErrorWithStatusCode('Invalid request: missing domains in the request data', STATUS_BAD_REQUEST);
+    }
+
+    data.domains.forEach((url) => {
+      if (!isValidUrl(url)) {
+        throw new ErrorWithStatusCode(`Invalid request: ${url} is not a valid domain`, STATUS_BAD_REQUEST);
+      }
+    });
+
+    if (!hasText(data.name)) {
+      throw new ErrorWithStatusCode('Invalid request: missing name in the request data', STATUS_BAD_REQUEST);
+    }
   }
 
-  // eslint-disable-next-line no-unused-vars
-  function validateImsOrgId(imsOrgId) {
-    // Not implemented yet
+  /**
+   * Validate the IMS Org ID.
+   * @param imsOrgId
+   * @param imsUserToken
+   */
+  function validateImsOrgId(imsOrgId, imsUserToken) {
+    if (!imsOrgId) {
+      throw new ErrorWithStatusCode('Missing x-ims-gw-org-id header', STATUS_UNAUTHORIZED);
+    }
+    const imsUserProfile = imsClient.getImsUserProfile(imsUserToken);
+    const { organizations } = imsUserProfile;
+    if (!organizations.includes(imsOrgId)) {
+      throw new ErrorWithStatusCode('Invalid request: Organization not found', STATUS_UNAUTHORIZED);
+    }
+  }
+
+  /**
+   * Get the IMS user token from the headers.
+   * @param headers
+   * @returns imsUserToken
+   */
+  function getImsUserToken(headers) {
+    const authorizationHeader = headers.authorization;
+    if (!authorizationHeader) {
+      throw new ErrorWithStatusCode('Missing Authorization header', STATUS_UNAUTHORIZED);
+    }
+    return authorizationHeader.replace('Bearer ', '');
```

Comment:
Instead of pulling this from the header ourselves, could we leverage the `profile` that has been set up for the request via the AdobeImsHandler? https://github.com/adobe/spacecat-shared/blob/main/packages/spacecat-shared-http-utils/src/auth/handlers/ims.js#L125-L131

Reply:
We need to pull the imsUserToken from the header as we validate the imsOrgId of the user here: https://github.com/adobe/spacecat-api-service/pull/569/files#diff-31f2b28cc5f3cd0e54aa7a6ab76c90dae8509fd1c1a8452712fa1822ca5bb2dbR96. To retrieve the list of organizations the user is affiliated with, we invoke the function `wait imsClient.getImsUserProfile(imsUserToken);` which needs the imsUserToken.